safe_transaction_service/history/indexers/element_already_processed_checker.py

```diff
@@ -0,0 +1,51 @@
+from logging import getLogger
+from typing import Optional
+
+from hexbytes import HexBytes
+
+from safe_transaction_service.utils.utils import FixedSizeDict
+
+logger = getLogger(__name__)
+
+
+class ElementAlreadyProcessedChecker:
+    def __init__(self):
+        self._processed_element_cache = FixedSizeDict(maxlen=40_000)  # Around 3MiB
+
+    def clear(self) -> None:
+        return self._processed_element_cache.clear()
+
+    def get_key(self, tx_hash: HexBytes, block_hash: HexBytes, index: int) -> HexBytes:
+        tx_hash = HexBytes(tx_hash)
+        block_hash = HexBytes(block_hash or 0)
+        index = HexBytes(index)
+        return tx_hash + block_hash + index
+
+    def is_processed(
+        self, tx_hash: HexBytes, block_hash: Optional[HexBytes], index: int = 0
+    ) -> bool:
+        tx_id = self.get_key(tx_hash, block_hash, index)
+        return tx_id in self._processed_element_cache
+
+    def mark_as_processed(
```

Comment:
Could you add comment to this function?
"""
Mark element as processed if weren't marked before

:return: True if... False if...